File: lab01/fib/fibiter.py
import sys

def fib(n):
  a,b = 0,1
  while n > 1:
    a,b,n = b,a+b,n-1
  return b

n=42
if len(sys.argv) > 1:
  n=int(sys.argv[1])
print (fib(n))

File: lab01/fib/fibiter.c
#include <stdio.h>
#include <stdlib.h>
int main(int argc, char *argv[]) {
	int a = 0, b = 1, n = 46, i = n;
	 if (argc > 1)
		i = n = atoi(argv[1]);
	 while (--i) {
		int t = a + b;
		a = b;
		b = t;
	}
	printf("fib(%d)=%d\n", n, b);
	return 0;
}
